Lib/typing.py

```diff
@@ -2667,3 +2668,23 @@ class re(metaclass=_DeprecatedType):
 
 re.__name__ = __name__ + '.re'
 sys.modules[re.__name__] = re
+
+
+def reveal_type(obj: T, /) -> T:
```

Comment:
Do we use type hints in `typing.py`? 🤔

Reply:
If this is merged, we will :). I don't see any reason not to, especially in cases like this where the type is simple and helps document the behavior.

Reply:
There are type annotations in a few lines of `typing.py`:

https://github.com/python/cpython/blob/270a09184d312856ca112396daec8f360cc5510e/Lib/typing.py#L2531

It's just about the only place in the stdlib in which I've spotted any type annotations other than importlib :)